src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs

```diff
@@ -302,23 +305,44 @@ internal void AddConnectionLevelDetailsToActivity(string dataSource, Activity sq
                     ConnectionDetailCache.TryAdd(dataSource, connectionDetails);
                 }
 
-                if (!string.IsNullOrEmpty(connectionDetails.ServerHostName))
-                {
-                    sqlActivity.SetTag(SemanticConventions.AttributeNetPeerName, connectionDetails.ServerHostName);
-                }
-                else
+                if (!string.IsNullOrEmpty(connectionDetails.InstanceName))
                 {
-                    sqlActivity.SetTag(SemanticConventions.AttributeNetPeerIp, connectionDetails.ServerIpAddress);
+                    sqlActivity.SetTag(SemanticConventions.AttributeDbMsSqlInstanceName, connectionDetails.InstanceName);
                 }
 
-                if (!string.IsNullOrEmpty(connectionDetails.InstanceName))
+                if (this.emitOldAttributes)
                 {
-                    sqlActivity.SetTag(SemanticConventions.AttributeDbMsSqlInstanceName, connectionDetails.InstanceName);
+                    if (!string.IsNullOrEmpty(connectionDetails.ServerHostName))
+                    {
+                        sqlActivity.SetTag(SemanticConventions.AttributeNetPeerName, connectionDetails.ServerHostName);
+                    }
+                    else
+                    {
+                        sqlActivity.SetTag(SemanticConventions.AttributeNetPeerIp, connectionDetails.ServerIpAddress);
+                    }
+
+                    if (!string.IsNullOrEmpty(connectionDetails.Port))
+                    {
+                        sqlActivity.SetTag(SemanticConventions.AttributeNetPeerPort, connectionDetails.Port);
+                    }
                 }
 
-                if (!string.IsNullOrEmpty(connectionDetails.Port))
+                // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/trace/semantic_conventions/http.md
+                if (this.emitNewAttributes)
                 {
-                    sqlActivity.SetTag(SemanticConventions.AttributeNetPeerPort, connectionDetails.Port);
+                    if (!string.IsNullOrEmpty(connectionDetails.ServerHostName))
+                    {
+                        sqlActivity.SetTag(SemanticConventions.AttributeServerAddress, connectionDetails.ServerHostName);
+                    }
+                    else
+                    {
+                        sqlActivity.SetTag(SemanticConventions.AttributeServerSocketAddress, connectionDetails.ServerIpAddress);
+                    }
+
+                    if (!string.IsNullOrEmpty(connectionDetails.Port))
```

Comment:
Could this be default port for SQL Server?
https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/trace/semantic_conventions/database.md#:~:text=%5B1%5D%3A%20If%20using%20a%20port%20other%20than%20the%20default%20port%20for%20this%20DBMS%20and%20if%20server.address%20is%20set

Reply:
The old code wasn't making any distinction between default port, so I didn't here.

The spec says 
> Conditionally Required if using a port other than the default port for this DBMS and if server.address is set.

The spec doesn't define a Requirement Level for the default case.

Reply:
Please add a todo for this to follow up. Needs some clarification from spec.